Add support for linux and possibly other OSes

furbrain opened the pull request:
Use SDL to provide the wrapper gui to simulate the tingbot
touches are offset when the wrapper is present
I believe the "linux" setup should work even under windows, even though I have not been able to test this - it simply uses pygame to simulate the tingbot hardware (in a similar way to the way the osx module uses Cocoa objects to emulate the tingbot)

furbrain committed:
Add support for linux and possibly other OSes

Use SDL to provide the wrapper gui to simulate the tingbot
touches are offset when the wrapper is present

Rob4001 commented:
Can confirm that the Wrapper class works for windows, but would suggest having all the wrapper stuff in the separated linux/windows platform_specific files similar to the osx file instead of having it in graphics.
could we have it so in _create_surface(), surface = platform_specific.fix_window();. It would allow more uniformity between the Operating System Stuff

furbrain committed:
Merge remote-tracking branch 'upstream/master'
Remove references to win.py

Conflicts:
	tingbot/platform_specific/__init__.py

furbrain committed:
move linux to sdl_wrapper - as it isn't specific to linux, and works in windows as well

furbrain commented:
@Rob4001 I think you're right - moving the Wrapper class into platform_specific is a lot neater. I'll commit that shortly

furbrain committed:
Move Wrapper and Button classes to sdl_wrappper.
Change platform_specific.fixup_window functionality to return pygame surface for Screen

furbrain commented:
@joerick We need to test that I have not broken osx functionality - I don't have access to Apple hardware

furbrain committed:
Tidy input.py

- simplified definition of hitarea
- create wrapper function to call function from mouse_* and give position translated
  to screen coordinates

furbrain committed:
Stop git from noticing text editor backup files

joerick commented:
This is looking really nice! I'll make some comments inline

joerick commented on tingbot/graphics.py:
```diff
@@ -187,9 +186,8 @@ def __init__(self):
-        pygame.init()
-        surface = pygame.display.set_mode((320, 240))
-        platform_specific.fixup_window()
+        from . import platform_specific
```
Wondering why this import has moved here and not at the top?

joerick replied:
Ah, I see, circular import

joerick commented on tingbot/platform_specific/osx.py:
```diff
@@ -136,6 +136,9 @@ def buttons(self):
 window_controller = None
 
 def fixup_window():
```
Now that this function is returning the pygame surface, can we rename `fixup_window` to `create_main_surface`?

furbrain replied:
Yes, good idea

joerick commented on tingbot/platform_specific/sdl_wrapper.py:
```diff
@@ -0,0 +1,66 @@
+        self.surface = pygame.display.set_mode((470, 353))
+        background = pygame.image.load(os.path.join(os.path.dirname(__file__), 'bot.png'))
+        self.surface.blit(background,(0,0))
+        self.screen = self.surface.subsurface((60,40,320,240))
```
This is neat! I didn't know about subsurfaces.

joerick commented on tingbot/platform_specific/sdl_wrapper.py:
```diff
@@ -0,0 +1,66 @@
+        for x in range(4):
+            self.buttons.append(Button(self.surface.subsurface(xPositions[x],0,22,12),x))
+
+    def get_screen(self):
```
Why the getter `get_screen` here?

furbrain replied:
I think this is a hangover from when I had things arranged differently. I can just access the screen directly, so I agree it is not needed.

Will fix when I get home tonight

furbrain committed:
Remove unnecessary getter

furbrain committed:
Refactor fixup_window -> create_main_surface

furbrain committed:
Refactor fixup_window -> create_main_surface

joerick commented on tingbot/graphics.py:
```diff
@@ -187,9 +186,8 @@ def __init__(self):
-        surface = pygame.display.set_mode((320, 240))
-        platform_specific.fixup_window()
+        from . import platform_specific
+        surface = platform_specific.fixup_window()
```
Just one lingering fixup_window left!

furbrain committed:
Merge branch 'master' of https://github.com/furbrain/tingbot-python

joerick commented on tingbot/platform_specific/osx.py:
```diff
@@ -135,7 +135,10 @@ def buttons(self):
 
-def fixup_window():
+def create_main_surface():
+    pygame.init()
```
Need to add an `import pygame` here

furbrain committed:
Add import pygame

furbrain commented:
I've made a simple demo script which allows you to easily check all the buttons and the touch screen and the run loop
Do you want this adding?

joerick commented:
Thanks @furbrain this is looking good.

joerick commented:
If you get a chance could you guys screenshot what the window looks like on Windows and Linux? I tried the sdl_wrapper on mac and got some graphical glitches, so wanted to see how it's looking on other platforms.

Rob4001 commented:
http://pasteboard.co/70MQ6RO.png
